CMakeLists.txt

```diff
@@ -1,78 +1,840 @@
+  "${library_root}/xml_arch.cc"
+)
+
+set(sleigh_source_list
+  "${library_root}/sleigh.cc"
+  "${library_root}/pcodeparse.cc"
+  "${library_root}/pcodecompile.cc"
+  "${library_root}/sleighbase.cc"
+  "${library_root}/slghsymbol.cc"
+  "${library_root}/slghpatexpress.cc"
+  "${library_root}/slghpattern.cc"
+  "${library_root}/semantics.cc"
+  "${library_root}/context.cc"
+  "${library_root}/filemanage.cc"
+)
+
+set(sleigh_ghidra_source_list
+  "${library_root}/ghidra_arch.cc"
+  "${library_root}/inject_ghidra.cc"
+  "${library_root}/ghidra_translate.cc"
+  "${library_root}/loadimage_ghidra.cc"
+  "${library_root}/typegrp_ghidra.cc"
+  "${library_root}/database_ghidra.cc"
+  "${library_root}/ghidra_context.cc"
+  "${library_root}/cpool_ghidra.cc"
+  "${library_root}/ghidra_process.cc"
+  "${library_root}/comment_ghidra.cc"
+  "${library_root}/string_ghidra.cc"
+)
+
+set(sleigh_slacomp_source_list
+  "${library_root}/slgh_compile.cc"
+  "${library_root}/slghparse.cc"
+  "${library_root}/slghscan.cc"
+)
+
+#
+# Locate the BFD library and its dependencies
+#
+
+# BFD
+find_path(bfd_include_path NAMES "bfd.h")
+if(NOT bfd_include_path)
+  message(SEND_ERROR "Failed to locate the bfd header file. Try to install the binutils development package")
+endif()
+
+find_library(bfd_library_path
+  NAMES
+    "libbfd.a"
+    "libbfd.so"
+)
+
+if(NOT bfd_library_path)
+  message(SEND_ERROR "Failed to locate the bfd library. Try to install the binutils development package")
+endif()
+
+add_library(thirdparty_bfd UNKNOWN IMPORTED GLOBAL)
+set_target_properties(thirdparty_bfd PROPERTIES
+  IMPORTED_LOCATION
+    "${bfd_library_path}"
 
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+  INTERFACE_INCLUDE_DIRECTORIES
+    "${bfd_include_path}"
+)
 
-include(prevent_in_source_builds)
-include(GNUInstallDirs)
-include(project_settings)
+target_link_libraries(thirdparty_bfd INTERFACE
+  ${CMAKE_DL_LIBS}
+)
+
+target_compile_definitions(thirdparty_bfd INTERFACE
+  PACKAGE=\"sleigh\"
+  PACKAGE_VERSION=\"${PROJECT_VERSION}\"
+)
+
+# ZLIB
+find_package(ZLIB REQUIRED)
+target_link_libraries(thirdparty_bfd INTERFACE
+  ZLIB::ZLIB
+)
+
+# iberty
+find_library(iberty_library_path
+  NAMES
+    "libiberty.a"
+    "libiberty.so"
+)
+
+if(NOT iberty_library_path)
+  message(SEND_ERROR "Failed to locate the iberty library. Try to install the libiberty development package")
+endif()
+
+add_library(thirdparty_iberty UNKNOWN IMPORTED GLOBAL)
+set_target_properties(thirdparty_iberty PROPERTIES
+  IMPORTED_LOCATION
+    "${iberty_library_path}"
+)
+
+target_link_libraries(thirdparty_bfd INTERFACE
+  thirdparty_iberty
+)
+
+#
+# sleigh settings
+#
 
-# Sleigh build settings
 add_library(sleigh_settings INTERFACE)
+target_compile_features(sleigh_settings INTERFACE
+  cxx_std_17
+)
+
+set_target_properties(sleigh_settings PROPERTIES
+  INTERFACE_POSITION_INDEPENDENT_CODE ON
+)
+
+if(SLEIGH_ENABLE_SANITIZERS)
+  message(STATUS "sleigh: Sanitizers have been enabled")
```

Comment:
I think there should be separate flags for ASan and UBSan.

Reply:
@tetsuo-cpp for vcpkg, at least, I think the way we have it is that ubsan and asan come as a pair. I'd prefer to just say "do sanitizers" and you get both. It think perhaps the point you're making is that it's desirable to have ubsan (e.g. in a release build) but not asan, though for all practical purposes, likely undesirable to have asan without ubsan.